### PR TITLE
feat: implement IsDirectory for OSS

### PR DESCRIPTION
**Pull request by shuangkun**
Implement IsDirectory for OSS

### CODE REVIEW

File: workflow/artifacts/oss/oss.go [hunk 1]
```diff
@@ -462,6 +462,21 @@ func ListOssDirectory(bucket *oss.Bucket, objectKey string) (files []string, err
...
 
-func (g *ArtifactDriver) IsDirectory(artifact *wfv1.Artifact) (bool, error) {
-	return false, errors.New(errors.CodeNotImplemented, "IsDirectory currently unimplemented for OSS")
+// IsDirectory tests if the key is acting like a OSS directory
+func (ossDriver *ArtifactDriver) IsDirectory(artifact *wfv1.Artifact) (bool, error) {
+	osscli, err := ossDriver.newOSSClient()
+	if err != nil {
+		return false, err
+	}
+	bucketName := artifact.OSS.Bucket
```
Comment (Joibel) [inline workflow/artifacts/oss/oss.go:471]: Not convinced we need `bucketName` and `objectName` as variables here as they're initialized as simple copies and used once.

**Review by Joibel — APPROVED** [review]
Looks good, and thanks for the contribution.

File: workflow/artifacts/oss/oss.go [hunk 2]
```diff
@@ -462,6 +462,21 @@ func ListOssDirectory(bucket *oss.Bucket, objectKey string) (files []string, err
...
+	bucket, err := osscli.Bucket(bucketName)
+	if err != nil {
+		return false, err
+	}
+	objectName := artifact.OSS.Key
+	isDir, err := IsOssDirectory(bucket, objectName)
+	if err != nil {
+		return false, err
```
[the remark this answers is not in the document]
Comment (shuangkun) [reply workflow/artifacts/oss/oss.go:479]: Do you mean there are some error cause "IsOssDirectory" return error. e.g. timeout or certification ？

File: workflow/artifacts/oss/oss.go (hunk 1 above)
Comment (shuangkun) [reply workflow/artifacts/oss/oss.go:471]: ok, although it used once. but i want to be consistent with  "ListObjects".

File: workflow/artifacts/oss/oss.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (shuangkun) [reply workflow/artifacts/oss/oss.go:479]: I will check it.

File: workflow/artifacts/oss/oss.go (hunk 1 above)
Comment (Joibel) [reply workflow/artifacts/oss/oss.go:471]: Ok, sounds reasonable.

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**